### PR TITLE
Fix typos and other doc/comment issues, add .idea to .gitignore, and …

### PR DESCRIPTION
**Pull request by xoltar**
…delete apparently unused function theano.compile.mode.check_equal.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,4 +37,5 @@ Theano.suo
 .ipynb_checkpoints
 .pydevproject
 .ropeproject
-core
\ No newline at end of file
+core
+.idea
```
Comment (nouiz) [inline .gitignore]: Out or curiosity, what system generate that?

File: theano/compile/mode.py [hunk 2]
```diff
@@ -18,35 +18,6 @@
 _logger = logging.getLogger('theano.compile.mode')
 
 
-def check_equal(x, y):
```
Comment (nouiz) [inline theano/compile/mode.py]: This cause a failure in our test as now, the numpy import isn't used anymore in that file. Can you remove the numpy import from that file?

/home/travis/build/Theano/Theano/theano/compile/mode.py:8:1: F401 'numpy' imported but unused

File: theano/compile/mode.py [hunk 3]
```diff
@@ -422,6 +393,7 @@ def get_mode(orig_string):
             # This might be required if the string is 'ProfileMode'
             from .profilemode import ProfileMode  # noqa
             from .profilemode import prof_mode_instance_to_print
+            # TODO: Can't we look up the name and invoke it rather than using eval here?
```
Comment (nouiz) [inline theano/compile/mode.py]: Probably. There is only the modes ProfileMode (deprecated, we can probably remove it now) and MonitorMode that aren't the default Mode.
Mostly, if we add the elif for MonitorMode and Mode and remove ProfileMode, then we could remove that eval.

**Comment by nouiz** [comment]
Thanks for the PR. I just one comment to fix before merging this PR as it make our test fail.

File: .gitignore (hunk 1 above)
Comment (xoltar) [inline .gitignore]: This is generated by the Intellij IDEA Python plugin, and probably PyCharm does the same.

**Comment by xoltar** [comment]
@nouiz Removed the numpy import.

File: theano/compile/function.py [hunk 4]
```diff
@@ -100,6 +100,9 @@ def function(inputs, outputs=None, mode=None, updates=None, givens=None,
         If True, do not perform any automatic update on Variables. If False
         (default), perform them all. Else, perform automatic updates on all
         Variables that are neither in "updates" nor in "no_default_updates".
+    accept_inplace : bool
+        True iff the graph can contain inplace operations prior to the
+        optimization phase (default is False).
```
Comment (nouiz) [inline theano/compile/function.py]: Can you add that this is not recomanded to use it and is not supported? It was for that reason that it wasn't documented.

File: theano/compile/function.py [hunk 5]
```diff
@@ -115,10 +118,10 @@ def function(inputs, outputs=None, mode=None, updates=None, givens=None,
         Ops in the graph, an Exception will be raised.
     allow_input_downcast: bool or None
         True means that the values passed as inputs when calling the function
-        can be silently downcasted to fit the dtype of the corresponding
+        can be silently down-casted to fit the dtype of the corresponding
```
Comment (nouiz) [inline theano/compile/function.py]: We use nowhere the version with a dash. Why did you change that? Maybe we are better to be consistent.

But otherwise, I don't have a strong preference for that one.

File: theano/compile/pfunc.py [hunk 6]
```diff
@@ -306,6 +306,9 @@ def pfunc(params, outputs=None, mode=None, updates=None, givens=None,
         If False (default), perform them all. Else, perform automatic updates
         on all Variables that are neither in "updates" nor in
         "no_default_updates".
+    accept_inplace : bool
+        True iff the graph can contain inplace operations prior to the
+        optimization phase (default is False).
```
Comment (nouiz) [inline theano/compile/pfunc.py]: idem

**Comment by nouiz** [comment]
Just a few small comment. thanks.

File: theano/compile/function.py (hunk 4 above)
Comment (xoltar) [inline theano/compile/function.py]: @nouiz Shall I remove the description and just say it's unsupported, or keep what's there and add the warning to the end?

File: theano/compile/function.py (hunk 5 above)
Comment (xoltar) [inline theano/compile/function.py]: @nouiz Just because the spell-checker didn't like the one without the dash. I can change it back if you like.

File: theano/compile/function.py (hunk 4 above)
Comment (lamblin) [inline theano/compile/function.py]: IMO, keep the description and add a warning that it is not recommended.
There are use cases where it can be useful, especially when cloning or unpickling graphs that are already optimized, but it is not the typical case.

File: theano/compile/function.py (hunk 5 above)
Comment (nouiz) [inline theano/compile/function.py]: My english isn't perferct. I have found downcast in dictionary by googling. I don't have a preference. @lamblin answered for the other comments. I like is suggestion.

Comment (lamblin) [inline theano/compile/function.py]: Either version is fine by me, I'd say don't bother changing it back if it makes the spell-checker happy.

**Comment by xoltar** [comment]
@nouiz Added the required warnings.

**Comment by nouiz** [comment]
thanks
